### DISCUSSION TITLE
OAuth authorization callback is not working where there is a # in the URL

### DISCUSSION THREAD

**Discussion by matthewchung74** [community/community#24755] [Apps, API and Webhooks]
<p>I am not sure if this is the right place for this, but the OAuth authorization callback is not working where there is a # in the URL.</p>
<p>Here is the callback url<br>
<img src="https://user-images.githubusercontent.com/1685700/181086585-4046466e-e1d3-43ff-b4bc-581cc9253f39.png" alt="Screen Shot 2021-05-19 at 10.00.03 AM" data-base62-sha1="3nu9zKAVul68JtOICQPL261OmLx" width="636" height="177"></p>
<p>which returns something like this. You can see the code query string showing before the path when there is a # in the path.</p>
<p><a href="http://localhost:9000/?code=CODE_FROM_GITHUB_SERVER#/signin" class="onebox" target="_blank" rel="noopener nofollow ugc">http://localhost:9000/?code=CODE_FROM_GITHUB_SERVER#/signin</a></p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>I think in this case we are parsing the hash mark (#) as a fragment identifier.</p>
<p>The fragment identifier introduced by a hash mark # is the optional last part of a URL for a document. It is typically used to identify a portion of that document (to scroll to a specific element id).</p>
<p>We rebuild the URI in accordance with RFC3986, where section 3 of RFC3986 indicates the fragment must be at the end of the URI:</p>
<blockquote>
<p>The generic URI syntax consists of a hierarchical sequence of<br>
components referred to as the scheme, authority, path, query, and<br>
fragment.</p>
</blockquote>
<blockquote>
<pre><code class="lang-auto"> URI  = scheme ":" hier-part [ "?" query ] [ "#" fragment ]
</code></pre>
</blockquote>
<p>Ref: <a href="https://tools.ietf.org/html/rfc3986">https://tools.ietf.org/html/rfc3986</a></p>
<p>So in this case we are rebuilding the URI, after adding the query parameter <code>code</code> and moving the fragment identifier to the end of the URI.</p>
<p>If that URL is not configurable in your app, you may have to find a way to set up a html redirect to this URL from one that doesn’t have a # character - so that we’re not adding the token in the middle.</p>
